chia/full_node/mempool_manager.py

```diff
@@ -455,6 +457,10 @@ async def validate_spend_bundle(
         removal_names_from_coin_spends: set[bytes32] = set()
         fast_forward_coin_ids: set[bytes32] = set()
         bundle_coin_spends: dict[bytes32, BundleCoinSpend] = {}
+        # Map of latest unspent ID to its puzzle hash and item's spends
+        fast_forward_unspents_map: dict[bytes32, tuple[bytes32, set[bytes32]]] = {}
+        # Map of singleton puzzle hash to its latest unspent linage info
+        singleton_latest_unspent_info_map: dict[bytes32, UnspentLineageInfo] = {}
```

Comment:
it would seem more reliable to make these mappings part of the sqlite database instead. then you wouldn't need to do so much accounting, to keep the data structures consistent

Reply:
Please see my answer below and let me know what you think. Unless there is a subtle bug, we're not doing much accounting, just updating the singletons latest versions info as we remove the old ones. This is similar to how we keep `bundle_coin_spends` outside the in-memory DB.